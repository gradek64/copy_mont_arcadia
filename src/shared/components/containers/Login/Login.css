/** @define Login */

.Login {
  box-sizing: border-box;
  margin-top: 28px;
  padding: 0 $page-padding;
}

@media $min-tablet {
  .Login {
    &--signin {
      & .Login {
        &-header {
          border-bottom: 1px solid $signin-title-border-color;
          padding-bottom: 20px;
        }
      }
    }
  }
}

@media $min-laptop {
  .Login {
    &--signin {
      & .Login {
        &-form {
          width: 60%;
        }
      }
    }
  }
}
