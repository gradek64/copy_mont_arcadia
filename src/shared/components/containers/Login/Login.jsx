import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { validate } from '../../../lib/validator'
import {
  sendAnalyticsClickEvent,
  sendAnalyticsErrorMessage,
  ANALYTICS_ERROR,
  GTM_CATEGORY,
  GTM_ACTION,
} from '../../../analytics'
import {
  setFormField,
  touchedFormField,
  resetForm,
  setFormMessage,
  touchedMultipleFormFields,
  focusedFormField,
} from '../../../actions/common/formActions'
import { loginRequest } from '../../../actions/common/authActions'
import Button from '../../common/Button/Button'
import Input from '../../common/FormComponents/Input/Input'
import Message from '../../common/FormComponents/Message/Message'
import Form from '../../common/FormComponents/Form/Form'
import { prop, mapObjIndexed, path, isEmpty } from 'ramda'
import Checkbox from '../../common/FormComponents/Checkbox/Checkbox'
import { isFeatureRememberMeEnabled } from '../../../../shared/selectors/featureSelectors'
import { isUserPartiallyAuthenticated } from '../../../../shared/selectors/userAuthSelectors'
import { getUserEmail } from '../../../selectors/common/accountSelectors'

const defaultFormName = 'login'

@connect(
  (state, props) => ({
    loginForm: state.forms[props.formName || defaultFormName],
    isFeatureRememberMeEnabled: isFeatureRememberMeEnabled(state),
    isUserPartiallyAuthenticated: isUserPartiallyAuthenticated(state),
    userEmail: getUserEmail(state),
    isPartiallyAuthenticated: isUserPartiallyAuthenticated(state),
  }),
  {
    setFormField,
    touchedFormField,
    resetForm,
    sendAnalyticsClickEvent,
    sendAnalyticsErrorMessage,
    setFormMessage,
    touchedMultipleFormFields,
    focusedFormField,
    loginUser: loginRequest,
  }
)
class Login extends Component {
  static propTypes = {
    loginForm: PropTypes.object.isRequired,
    loginUser: PropTypes.func.isRequired,
    setFormField: PropTypes.func.isRequired,
    touchedFormField: PropTypes.func.isRequired,
    resetForm: PropTypes.func.isRequired,
    setFormMessage: PropTypes.func.isRequired,
    touchedMultipleFormFields: PropTypes.func.isRequired,
    focusedFormField: PropTypes.func.isRequired,
    getNextRoute: PropTypes.func,
    className: PropTypes.string,
    successCallback: PropTypes.func,
    errorCallback: PropTypes.func,
    formName: PropTypes.string,
    userEmail: PropTypes.string,
    isPartiallyAuthenticated: PropTypes.bool,
  }

  static defaultProps = {
    className: '',
    isPartiallyAuthenticated: false,
    formName: defaultFormName,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  componentDidMount() {
    this.resetForm()
  }

  componentDidUpdate(prevProps) {
    const {
      focusedFormField,
      loginForm,
      userEmail,
      isFeatureRememberMeEnabled,
      formName,
    } = this.props

    if (isFeatureRememberMeEnabled && prevProps.userEmail !== userEmail) {
      this.updateUserEmail()
    }

    if (isFeatureRememberMeEnabled !== prevProps.isFeatureRememberMeEnabled) {
      this.resetForm()
    }

    const { isFocused } = loginForm.fields.password
    const wasFocused = prevProps.loginForm.fields.password.isFocused
    if (isFocused && wasFocused !== isFocused) {
      focusedFormField(formName, 'password', false)
    }
  }

  componentWillUnmount() {
    this.resetForm()
  }

  onSubmit = (e) => {
    e.preventDefault()
    const {
      touchedMultipleFormFields,
      loginUser,
      getNextRoute,
      loginForm,
      successCallback,
      errorCallback,
      formName,
      sendAnalyticsClickEvent,
      sendAnalyticsErrorMessage,
    } = this.props
    touchedMultipleFormFields(formName, Object.keys(loginForm.fields))
    if (Object.keys(this.getErrors()).length) {
      sendAnalyticsErrorMessage(ANALYTICS_ERROR.LOGIN_FAILED)
      return
    }

    const {
      email,
      password,
      // Temporarily set default value for `rememberMe` incase it is not enabled
      rememberMe = { value: false },
    } = loginForm.fields

    sendAnalyticsClickEvent({
      category: GTM_CATEGORY.SIGN_IN,
      action: GTM_ACTION.SUBMIT,
      label: 'sign-in-button',
      rememberMe: !!rememberMe.value,
    })

    loginUser({
      credentials: {
        username: email.value,
        password: password.value,
        rememberMe: !!rememberMe.value,
      },
      getNextRoute,
      successCallback,
      errorCallback,
      formName,
    })
  }

  getErrors = () => {
    const validationSchema = {
      email: 'email',
      password: 'password',
      rememberMe: 'rememberMe',
    }
    return validate(
      validationSchema,
      mapObjIndexed(prop('value'), this.props.loginForm.fields),
      this.context.l
    )
  }

  updateUserEmail = () => {
    const {
      setFormField,
      isFeatureRememberMeEnabled,
      userEmail,
      formName,
    } = this.props

    if (!isFeatureRememberMeEnabled) {
      return
    }

    setFormField(formName, 'email', userEmail)
  }

  resetForm = () => {
    const {
      resetForm,
      setFormMessage,
      isFeatureRememberMeEnabled,
      userEmail,
      formName,
    } = this.props

    setFormMessage(formName, {})

    const formValues = {
      email: '',
      password: '',
    }

    if (isFeatureRememberMeEnabled) {
      const userRememberMe = this.getRememberMe()
      formValues.email = userRememberMe ? userEmail : ''
      formValues.rememberMe = userRememberMe
    }

    resetForm(formName, formValues)
  }

  // When 'Remember Me' is enabled
  //   Pre-fill the email address for a remembered user but don't force it to persist
  //   once the user edits the email field.
  // Otherwise
  //   Pass the email field through unaltered.
  getPreferredEmail(email) {
    if (this.props.isFeatureRememberMeEnabled) {
      const userRememberMe = this.getRememberMe()
      return userRememberMe && email.value === '' && !email.isDirty
        ? { ...email, value: this.props.userEmail }
        : email
    }

    return email
  }

  // When 'Remember Me' is enabled
  //   Pre-fill the 'Remember me' checkbox for a remembered user but don't force it
  //   to persist once the user toggles the checkbox.
  // Otherwise
  //   "Don't care" value, because it should only be used under this feature flag.
  getPreferredRememberMe(rememberMe) {
    if (this.props.isFeatureRememberMeEnabled) {
      const userRememberMe = this.getRememberMe()
      return userRememberMe && rememberMe.value === false && !rememberMe.isDirty
        ? { ...rememberMe, value: userRememberMe }
        : rememberMe
    }

    return undefined
  }

  getRememberMe() {
    return (
      Boolean(this.props.isFeatureRememberMeEnabled) &&
      Boolean(this.props.isUserPartiallyAuthenticated)
    )
  }

  render() {
    const { l } = this.context
    const {
      loginForm,
      formName,
      setFormField,
      touchedFormField,
      className,
      isFeatureRememberMeEnabled,
      isPartiallyAuthenticated,
    } = this.props
    const { email, password, rememberMe } = loginForm.fields
    const isPasswordFocused = password.isFocused
    const errors = this.getErrors()
    const message = path(['message', 'message'], loginForm)
    const setField = (field) => (e) =>
      setFormField(formName, field, e.target.value)
    const setChecked = (field) => (e) =>
      setFormField(formName, field, e.target.checked)
    const touchedField = (field) => () => touchedFormField(formName, field)

    const preferredEmail = this.getPreferredEmail(email)
    const preferredRememberMe = this.getPreferredRememberMe(rememberMe)

    return (
      <section className={`Login ${className}`}>
        <h3 className="Login-header">{l`Sign In`}</h3>
        <Form className="Login-form" onSubmit={this.onSubmit} method="POST">
          <Input
            id="Login-email"
            field={preferredEmail}
            name="email"
            type="email"
            errors={errors}
            label={l`Email address`}
            placeholder={l`example@domain.com`}
            setField={setField}
            touchedField={touchedField}
            autocomplete={isPartiallyAuthenticated ? 'off' : undefined}
            isRequired
          />

          <Input
            id="Login-password"
            field={password}
            name="password"
            type="password"
            errors={errors}
            label={l`Password`}
            setField={setField}
            touchedField={touchedField}
            isFocused={isPasswordFocused}
            autocomplete={isPartiallyAuthenticated ? 'on' : undefined}
            isRequired
          />

          {isFeatureRememberMeEnabled && (
            <Checkbox
              className="Checkbox-remember-me"
              checked={preferredRememberMe}
              onChange={setChecked('rememberMe')}
              name="rememberMe"
            >{l`Remember me`}</Checkbox>
          )}

          <Button
            className="Login-submitButton"
            isActive={isEmpty(errors)}
            isDisabled={!isEmpty(errors)}
            type="submit"
          >{l`Sign In`}</Button>
          {message && (
            <Message
              message={message}
              type={path(['message', 'type'], loginForm)}
            />
          )}
        </Form>
      </section>
    )
  }
}

export default Login
