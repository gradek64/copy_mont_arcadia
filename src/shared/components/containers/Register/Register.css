/** @define Register */

.Register {
  box-sizing: border-box;
  margin-top: 28px;
  padding: 0 $page-padding;
  &--guest {
    padding: 0;
    margin: 0;
  }
}

@media $min-tablet {
  .Register {
    &--signin {
      & .Register {
        &-header {
          border-bottom: 1px solid $signin-title-border-color;
          padding-bottom: 20px;
        }
      }
    }
    &--guest {
      max-width: 360px;
    }
  }
}

@media $min-laptop {
  .Register {
    &--signin {
      & .Register {
        &-form {
          width: 60%;
        }
      }
    }
  }
}
