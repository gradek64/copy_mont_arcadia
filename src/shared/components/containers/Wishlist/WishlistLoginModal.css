/** @define WishlistLoginModal */

.WishlistLoginModal {
  &-forgetPassword {
    margin-bottom: 10px;
  }

  &-forms {
    display: flex;
    flex-direction: column;
  }

  &-header {
    margin-bottom: 20px;
  }

  &-heartIcon {
    display: block;
    background-image: $wishlist-header-icon;
    background-repeat: no-repeat;
    background-size: contain;
    background-position: center;
    margin: 10px auto;
    width: 40px;
    height: 36px;
  }

  &-loginForm,
  &-register {
    margin-bottom: 25px;
    margin-top: 0;
  }

  &-text {
    font-size: 18px;
    line-height: 1.22;
    text-align: center;
  }
}

@media $min-tablet {
  .WishlistLoginModal {
    &-formContainer {
      width: 50%;
      overflow:hidden;
    }

    &-forms {
      flex-direction: row;
      justify-content: center;
    }

    &-header {
      align-items: center;
      display: flex;
      flex-direction: row;
      justify-content: center;
    }

    &-heartIcon {
      margin: 0 20px;
      width: 23px;
      height: 21px;
    }
  }
}

@media $min-laptop {
  .WishlistLoginModal {
    &-formContainer {
      padding: 0 50px;
    }
  }
}
