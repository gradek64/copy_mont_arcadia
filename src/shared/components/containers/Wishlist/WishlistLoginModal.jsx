import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Login from '../Login/Login'
import Register from '../Register/Register'
import ForgetPassword from '../ForgetPassword/ForgetPassword'

import {
  addToWishlistAfterLogin,
  captureWishlistEvent,
  getPaginatedWishlist,
} from '../../../actions/common/wishlistActions'
import {
  closeModal,
  clearModalChildren,
} from '../../../actions/common/modalActions'

import { getRoutePathWithParams } from '../../../selectors/routingSelectors'

if (process.browser) {
  require('./WishlistLoginModal.css')
}

@connect(
  (state) => ({
    relativePath: getRoutePathWithParams(state),
    isModalCancelled: state.modal.cancelled,
    pendingProductId: state.wishlist.pendingProductId,
  }),
  {
    addToWishlistAfterLogin,
    closeModal,
    getPaginatedWishlist,
    clearModalChildren,
    captureWishlistEvent,
  }
)
class WishlistLoginModal extends Component {
  static propTypes = {
    modifier: PropTypes.string,
    relativePath: PropTypes.string,
    pendingProductId: PropTypes.number,
    addToWishlistAfterLogin: PropTypes.func.isRequired,
    captureWishlistEvent: PropTypes.func,
    afterAddToWishlist: PropTypes.func,
    onCancelLogin: PropTypes.func,
    productDetails: PropTypes.object.isRequired,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  componentDidMount() {
    const { modifier, captureWishlistEvent } = this.props

    captureWishlistEvent('GA_WISHLIST_MODAL_PAGE_VIEW', { modifier })
  }

  componentWillUnmount() {
    const {
      modifier,
      isModalCancelled,
      captureWishlistEvent,
      onCancelLogin,
    } = this.props
    if (isModalCancelled) {
      captureWishlistEvent('GA_WISHLIST_MODAL_CLOSE', { modifier })
      if (onCancelLogin) onCancelLogin()
    }
  }

  handleSuccess = (authType) => {
    const { modifier, captureWishlistEvent } = this.props

    if (authType) {
      const action = `GA_WISHLIST_${authType.toUpperCase()}_SUCCESS`
      captureWishlistEvent(action)
    }

    switch (modifier) {
      case 'wishlist':
        return this.wishlistPageHandler()
      case 'plp':
      case 'pdp':
      case 'quickview':
      case 'minibag':
      case 'bundle':
        return this.addToWishlistHandler()
      default:
        return this.loginUserHandler()
    }
  }

  handleError = (authType) => {
    const { captureWishlistEvent } = this.props
    const action = `GA_WISHLIST_${authType.toUpperCase()}_ERROR`

    captureWishlistEvent(action)
  }

  loginUserHandler = () => {
    const { closeModal, clearModalChildren } = this.props
    const actions = [closeModal(), clearModalChildren()]
    return Promise.all([actions])
  }

  wishlistPageHandler = () => {
    const { closeModal, clearModalChildren, getPaginatedWishlist } = this.props
    const actions = [closeModal(), clearModalChildren(), getPaginatedWishlist()]
    return Promise.all([actions])
  }

  addToWishlistHandler = () => {
    const {
      addToWishlistAfterLogin,
      afterAddToWishlist,
      clearModalChildren,
      closeModal,
      modifier,
      pendingProductId,
      productDetails,
    } = this.props
    const actions = [clearModalChildren(), closeModal()]

    return addToWishlistAfterLogin(
      pendingProductId,
      modifier,
      productDetails
    ).then(() => {
      return Promise.all(actions).then(() => {
        if (afterAddToWishlist) afterAddToWishlist()
      })
    })
  }

  render() {
    const { l } = this.context
    const { relativePath } = this.props

    return (
      <div className="WishlistLoginModal">
        <div className="WishlistLoginModal-header">
          <i className="WishlistLoginModal-heartIcon" />
          <header className="WishlistLoginModal-text">{l`Sign in to save products to your Wishlist`}</header>
        </div>
        <div className="WishlistLoginModal-forms">
          <div className="WishlistLoginModal-formContainer">
            <Login
              className="WishlistLoginModal-loginForm"
              successCallback={() => this.handleSuccess('login')}
              errorCallback={() => this.handleError('login')}
              formName="wishlistLoginModal"
            />
            <ForgetPassword className="WishlistLoginModal-forgetPassword" />
          </div>
          <div className="WishlistLoginModal-formContainer">
            <Register
              className="WishlistLoginModal-register"
              source="WISHLIST"
              successCallback={() => this.handleSuccess('register')}
              errorCallback={() => this.handleError('register')}
              getNextRoute={() => relativePath}
              formName="wishlistLoginModal"
            />
          </div>
        </div>
      </div>
    )
  }
}

export default WishlistLoginModal
