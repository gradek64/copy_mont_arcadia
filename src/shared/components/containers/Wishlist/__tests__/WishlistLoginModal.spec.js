import testComponentHelper from 'test/unit/helpers/test-component'

import WishlistLoginModal from '../WishlistLoginModal'
import Login from '../../Login/Login'
import Register from '../../Register/Register'

describe('<WishlistLoginModal/>', () => {
  const renderComponent = testComponentHelper(
    WishlistLoginModal.WrappedComponent
  )
  const initialProps = {
    brandName: 'topshop',
    relativePath: '/search/?q=red',
    addToWishlistAfterLogin: jest.fn(() => Promise.resolve()),
    closeModal: jest.fn(() => Promise.resolve()),
    clearModalChildren: jest.fn(() => Promise.resolve()),
    getPaginatedWishlist: jest.fn(() => Promise.resolve()),
    wishlistPageCloseModal: jest.fn(() => Promise.resolve()),
    showModal: jest.fn(() => Promise.resolve()),
    afterAddToWishlist: jest.fn(),
    isModalCancelled: true,
    captureWishlistEvent: jest.fn(),
  }

  describe('@Renders', () => {
    it('in its default state', () => {
      const { getTree } = renderComponent(initialProps)
      expect(getTree()).toMatchSnapshot()
    })

    it('should render <Login/> and <Register/> components', () => {
      const { wrapper } = renderComponent(initialProps)
      expect(wrapper.find(Login).length).toBe(1)
      expect(wrapper.find(Register).length).toBe(1)
    })
  })
  describe('@lifecycle', () => {
    beforeEach(() => jest.clearAllMocks())

    describe('onComponentDidMount', () => {
      it('should capture modal page view', () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'plp',
        })
        instance.componentDidMount()
        expect(instance.props.captureWishlistEvent).toHaveBeenCalledWith(
          'GA_WISHLIST_MODAL_PAGE_VIEW',
          { modifier: 'plp' }
        )
      })
    })
    describe('componentWillUnmount', () => {
      beforeEach(() => jest.clearAllMocks())

      it('should not capture event if wishlist not cancelled by user', () => {
        const { instance } = renderComponent({
          ...initialProps,
          isModalCancelled: false,
        })
        instance.componentWillUnmount()
        expect(instance.props.captureWishlistEvent).toHaveBeenCalledTimes(0)
      })

      it('should capture modal close by user', () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'plp',
        })
        instance.componentWillUnmount()
        expect(instance.props.captureWishlistEvent).toHaveBeenCalledWith(
          'GA_WISHLIST_MODAL_CLOSE',
          { modifier: 'plp' }
        )
      })

      it('should invoke onCancelLogin callback', () => {
        const { instance } = renderComponent({
          ...initialProps,
          onCancelLogin: jest.fn(),
        })
        expect(instance.props.onCancelLogin).toHaveBeenCalledTimes(0)
        instance.componentWillUnmount()
        expect(instance.props.onCancelLogin).toHaveBeenCalledTimes(1)
      })

      it('should not invoke onCancelLogin callback if modal not cancelled by user', () => {
        const { instance } = renderComponent({
          ...initialProps,
          isModalCancelled: false,
          onCancelLogin: jest.fn(),
        })
        expect(instance.props.onCancelLogin).toHaveBeenCalledTimes(0)
        instance.componentWillUnmount()
        expect(instance.props.onCancelLogin).toHaveBeenCalledTimes(0)
      })
    })
  })
  describe('@Events', () => {
    describe('On success callback', () => {
      beforeEach(() => {
        jest.clearAllMocks()
      })
      it('should capture wishlist modal login success', () => {
        const { instance } = renderComponent(initialProps)
        instance.handleSuccess('login')
        expect(instance.props.captureWishlistEvent).toHaveBeenCalledWith(
          'GA_WISHLIST_LOGIN_SUCCESS'
        )
      })
      it('should capture wishlist modal register success', () => {
        const { instance } = renderComponent(initialProps)
        instance.handleSuccess('register')
        expect(instance.props.captureWishlistEvent).toHaveBeenCalledWith(
          'GA_WISHLIST_REGISTER_SUCCESS'
        )
      })
      it('clear and close modal if no modifier passed', async () => {
        const { instance } = renderComponent(initialProps)
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, add to wishlist if on plp', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'plp',
        })
        expect(instance.props.addToWishlistAfterLogin).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.addToWishlistAfterLogin).toHaveBeenCalledTimes(1)
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, add to wishlist if on pdp', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'pdp',
        })
        expect(instance.props.addToWishlistAfterLogin).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.addToWishlistAfterLogin).toHaveBeenCalledTimes(1)
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, add to wishlist if on quickview', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'quickview',
        })
        expect(instance.props.addToWishlistAfterLogin).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, add to wishlist if on plp', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'plp',
        })
        expect(instance.props.addToWishlistAfterLogin).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.addToWishlistAfterLogin).toHaveBeenCalledTimes(1)
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, add to wishlist if on pdp', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'pdp',
        })
        expect(instance.props.addToWishlistAfterLogin).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.addToWishlistAfterLogin).toHaveBeenCalledTimes(1)
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, add to wishlist if on minibag', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'minibag',
        })
        expect(instance.props.addToWishlistAfterLogin).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.addToWishlistAfterLogin).toHaveBeenCalledTimes(1)
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, add to wishlist if on quickview', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'quickview',
        })
        expect(instance.props.addToWishlistAfterLogin).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.addToWishlistAfterLogin).toHaveBeenCalledTimes(1)
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
      })
      it('clear and close modal, fetch wishlist items if on wishlist page', async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'wishlist',
        })
        expect(instance.props.closeModal).not.toHaveBeenCalled()
        expect(instance.props.clearModalChildren).not.toHaveBeenCalled()
        expect(instance.props.getPaginatedWishlist).not.toHaveBeenCalled()
        await instance.handleSuccess('login')
        expect(instance.props.closeModal).toHaveBeenCalledTimes(1)
        expect(instance.props.clearModalChildren).toHaveBeenCalledTimes(1)
        expect(instance.props.getPaginatedWishlist).toHaveBeenCalledTimes(1)
      })
      it("should call wishlist page close modal if logged in from the wishlist page'", async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'wishlist',
        })
        instance.wishlistPageHandler = jest.fn(() => Promise.resolve())
        expect(instance.wishlistPageHandler).not.toHaveBeenCalled()
        await instance.handleSuccess()
        expect(instance.wishlistPageHandler).toHaveBeenCalledTimes(1)
      })
      it("should call 'afterAddToWishlist' if provided", async () => {
        const { instance } = renderComponent({
          ...initialProps,
          modifier: 'minibag',
        })

        expect(instance.props.afterAddToWishlist).toHaveBeenCalledTimes(0)
        await instance.handleSuccess()
        expect(instance.props.afterAddToWishlist).toHaveBeenCalledTimes(1)
      })
    })
    describe('On error callback', () => {
      beforeEach(() => {
        jest.clearAllMocks()
      })
      it('should capture wishlist modal login error', () => {
        const { instance } = renderComponent(initialProps)
        instance.handleError('login')
        expect(instance.props.captureWishlistEvent).toHaveBeenCalledWith(
          'GA_WISHLIST_LOGIN_ERROR'
        )
      })
      it('should capture wishlist modal register error', () => {
        const { instance } = renderComponent(initialProps)
        instance.handleError('register')
        expect(instance.props.captureWishlistEvent).toHaveBeenCalledWith(
          'GA_WISHLIST_REGISTER_ERROR'
        )
      })
    })
  })
})
