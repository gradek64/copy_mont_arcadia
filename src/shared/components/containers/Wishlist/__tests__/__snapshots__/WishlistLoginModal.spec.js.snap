// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WishlistLoginModal/> @Renders in its default state 1`] = `
<div
  className="WishlistLoginModal"
>
  <div
    className="WishlistLoginModal-header"
  >
    <i
      className="WishlistLoginModal-heartIcon"
    />
    <header
      className="WishlistLoginModal-text"
    >
      Sign in to save products to your Wishlist
    </header>
  </div>
  <div
    className="WishlistLoginModal-forms"
  >
    <div
      className="WishlistLoginModal-formContainer"
    >
      <Connect(Login)
        className="WishlistLoginModal-loginForm"
        errorCallback={[Function]}
        formName="wishlistLoginModal"
        successCallback={[Function]}
      />
      <Connect(ForgetPassword)
        className="WishlistLoginModal-forgetPassword"
      />
    </div>
    <div
      className="WishlistLoginModal-formContainer"
    >
      <Connect(Register)
        className="WishlistLoginModal-register"
        errorCallback={[Function]}
        formName="wishlistLoginModal"
        getNextRoute={[Function]}
        source="WISHLIST"
        successCallback={[Function]}
      />
    </div>
  </div>
</div>
`;
